shared/src/test/scala/FetchSpec.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016-2018 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fetch
+
+import org.scalatest.{AsyncFreeSpec, Matchers, DoNotDiscover}
+
+import scala.concurrent._
+import java.util.concurrent._
+import scala.concurrent.duration._
+
+import cats.effect._
+
+@DoNotDiscover
+class FetchSpec extends AsyncFreeSpec with Matchers {
+  override val executionContext: ExecutionContext = ExecutionContext.Implicits.global
+  implicit val timer: Timer[IO] = IO.timer(executionContext)
+  implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+
+  def countFetches(r: Request): Int =
+    r.request match {
+      case FetchOne(_, _)       => 1
```

Comment:
Formatting seems off in a few match cases here. Are we enforcing auto formatting with Scalafmt or similar?

Reply:
Not right now, I can open a separate PR for adding scalafmt and not polluting this one